server/src/main/java/org/opensearch/search/startree/filter/provider/BoolStarTreeFilterProvider.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree.filter.provider;
+
+import org.opensearch.index.mapper.CompositeDataCubeFieldType;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.startree.filter.DimensionFilter;
+import org.opensearch.search.startree.filter.DimensionFilterMergerUtils;
+import org.opensearch.search.startree.filter.StarTreeFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Converts {@link BoolQueryBuilder} into {@link StarTreeFilter}
+ */
+public class BoolStarTreeFilterProvider implements StarTreeFilterProvider {
+
+    private static final Set<Class<? extends QueryBuilder>> SUPPORTED_NON_BOOL_QUERIES = Set.of(
+        TermQueryBuilder.class,
+        TermsQueryBuilder.class,
+        RangeQueryBuilder.class
+    );
+
+    @Override
+    public StarTreeFilter getFilter(SearchContext context, QueryBuilder rawFilter, CompositeDataCubeFieldType compositeFieldType)
+        throws IOException {
+        return processBoolQuery((BoolQueryBuilder) rawFilter, context, compositeFieldType);
+    }
+
+    private StarTreeFilter processBoolQuery(
+        BoolQueryBuilder boolQuery,
+        SearchContext context,
+        CompositeDataCubeFieldType compositeFieldType
+    ) throws IOException {
+        if (boolQuery.hasClauses() == false) {
+            return null;
```

Comment:
What does returning `null` denote? The filter can't be applied?

Reply:
It just indicates aggregator to fallback to default implementation and cannot be done via star tree